### PR TITLE
Unit tests for RegionRepository and RemoteDataSource

### LINKED ISSUE #53: Unit tests for RegionRepository
Based on tests from Antonio's project

### PR DESCRIPTION
**Pull request by jmgjmg**
Unit tests for RegionRepository and integration tests for RemoteDataSource

# New Pull Request
### Description
Unit tests for RegionRepositoryImpl based on Antonio's tests in his project. Bug found and fixed in RegionRepositoryImpl.getUserLanguage()

Integration tests of RemoteDataSourceImpl. Use MockWebser for Unit/Integration tests (no instrumentation test required). Validate that repository receives the right responses/errors from Arasaac API.

### Fixes
#53 and #54

### CODE REVIEW

File: app/src/test/java/com/architectcoders/aacboard/datasource/remote/RemoteDataSourceImplTest.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    @After
+    fun tearDown() {
+        server.shutdown()
+    }
+
+
+    @Test
+    fun `no pictogram found, http error 404, NoMathFound error returned`(): Unit = runTest {
```
Comment (cjimenezsanchez) [inline app/src/test/java/com/architectcoders/aacboard/datasource/remote/RemoteDataSourceImplTest.kt:50]: We haven't talk about this in the group but we could follow the GIVEN WHEN THEN approach for the tests naming if you like or not. Opened for discussion.

File: data/src/test/java/com/architectcoders/aacboard/data/repository/RegionRepositoryImplTest.kt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import org.mockito.junit.MockitoJUnitRunner
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+
+
+@RunWith(MockitoJUnitRunner::class)
+class RegionRepositoryImplTest {
+
+    @Test
+    fun `Returns default value when permission not granted`(): Unit = runBlocking {
```
Comment (cjimenezsanchez) [inline data/src/test/java/com/architectcoders/aacboard/data/repository/RegionRepositoryImplTest.kt:20]: Can't we use `runTest` instead of `runBlocking`?

**Comment by jmgjmg** [comment]
Changed from runBlocking to runTest

File: app/src/test/java/com/architectcoders/aacboard/datasource/remote/RemoteDataSourceImplTest.kt (hunk 1 above)
Comment (jmgjmg) [reply app/src/test/java/com/architectcoders/aacboard/datasource/remote/RemoteDataSourceImplTest.kt:50]: Updated in #57 trying to fit test method's names within detekt maximum line length

File: data/src/test/java/com/architectcoders/aacboard/data/repository/RegionRepositoryImplTest.kt (hunk 2 above)
Comment (jmgjmg) [reply data/src/test/java/com/architectcoders/aacboard/data/repository/RegionRepositoryImplTest.kt:20]: Changed

**Comment by jmgjmg** [comment]
Closed since changes integrated in #57

**Pull request closed without merging**